### PR TITLE
[FLINK-26825] Document State Machine Graph for JM Deployment

### PR DESCRIPTION
**Pull request by bgeng777**
- Add State Machine Graph for JM Deployment and document the meaning of each possible transition.
- The graph is attached here:
JM_deployment_state_machine

### CODE REVIEW

File: docs/content/docs/concepts/architecture.md [hunk 1]
```diff
@@ -45,6 +45,25 @@ The CR can be (re)applied on the cluster any time. The Operator makes continuous
 
 The Operator is built with the [Java Operator SDK](https://github.com/java-operator-sdk/java-operator-sdk) and uses the [Native Kubernetes Integration](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/resource-providers/native_kubernetes/) for launching Flink deployments and submitting jobs under the hood. The Java Operator SDK is a higher level framework and related tooling to support writing Kubernetes Operators in Java. Both the Java Operator SDK and Flink's native kubernetes integration itself is using the [Fabric8 Kubernetes Client](https://github.com/fabric8io/kubernetes-client) to interact with the Kubernetes API Server.
 
+## State Machine of JobManager Deployment
+The Operator will manage the lifecycle of JobManager Deployment. Its state machine is as follows:
```
Comment (mbalassi) [inline docs/content/docs/concepts/architecture.md:49]: nit: The Operator _manages_ the lifecycle of _the_ JobManager Deployment

File: docs/content/docs/concepts/architecture.md [hunk 2]
```diff
@@ -45,6 +45,25 @@ The CR can be (re)applied on the cluster any time. The Operator makes continuous
...
+
+{{< img src="/img/concepts/JM_deployment_state_machine.svg" alt="State Machine of JobManager Deployment" >}}
+
+The possible transitions usually indicate that there are some underlying changes:
+
+1. `MISSING` -> `DEPLOYING`: A new JM deployment exists and is being created
+2. `DEPLOYING` -> `DEPLOYED_NOT_READY`: The JM deployment exists and pass checks of the availability of replicas and JM port connectivity. Now, it is waiting the REST service to be ready
```
Comment (mbalassi) [inline docs/content/docs/concepts/architecture.md:56]: nit: ... and _passes the JM availability and connectivity checks_

**Review by mbalassi — COMMENTED** [review]
Thanks @bgeng777, added minor suggestions.

**Comment by bgeng777** [comment]
Thanks for the review. @mbalassi I have updated. PTAL.

**Pull request merged**